src/governance/governance.cpp

```diff
@@ -917,14 +947,15 @@ bool CGovernanceManager::ProcessVoteAndRelay(const CGovernanceVote& vote, CGover
 
 bool CGovernanceManager::ProcessVote(CNode* pfrom, const CGovernanceVote& vote, CGovernanceException& exception, CConnman& connman)
 {
-    ENTER_CRITICAL_SECTION(cs);
+    AssertLockNotHeld(cs_store);
+    ENTER_CRITICAL_SECTION(cs_store);
```

Comment:
~can't we just LOCK here now instead of enter / leave?~ never mind; not quite due to https://github.com/dashpay/dash/pull/6849/files#diff-6787065e28efe13420b2f5e85125894595bec4c84b0b917d952dc86abd879033R979

Would you like to add a TODO to address this?

Reply:
The `{ENTER,LEAVE}_CRITICAL_SECTION` usage is from [dash#1541](https://github.com/dashpay/dash/pull/1541) to resolve a deadlock issue.